dali/test/python/checkpointing/test_dali_checkpointing.py

```diff
@@ -287,6 +288,90 @@ def pipeline():
     compare_pipelines(p, restored, 1, 20)
 
 
+@dataclass
+class BaseDecoderConfig:
+    shard_id: int
+    num_shards: int
+    stick_to_shard: bool
+    pad_last_batch: bool
+    random_shuffle: bool
+
+@dataclass
+class VideoConfig:
+    sequence_length : int
+    stride : int
+    step : int
+
+
+@cartesian_params(
+    (0, 1, 3),
```

Comment:
How about a case that resumes from a non zero epoch and a non zero iteration?

Reply:
Isn't there one? I choose `num_epochs` from `(0, 1, 3)` and `iters_into_epoch` from `(0, 2)`, so there will be tests with `num_epochs=1, iters_into_epoch=2` and `num_epochs=3, iters_into_epoch=2`

Reply:
Oh, of course, sorry for the confusion. I somehow misread the suite as if the first three tuples where three different cases for `num_epochs, batch_size, iters_into_epoch` times the decoder times the video setups.